tests/core/test_timer.py

```diff
@@ -0,0 +1,110 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Getting Things GNOME! - a personal organizer for the GNOME desktop
+# Copyright (c) 2008-2015 - Lionel Dricot & Bertrand Rousseau
+#
+# This program is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program.  If not, see <http://www.gnu.org/licenses/>.
+# -----------------------------------------------------------------------------
+
+from unittest import TestCase
+from unittest.mock import patch
+
+from GTG.core.timer import Timer
+
+
+class Config(object):
+    # Creting a duck-type object Config
+    # to be able to create a Timer object later
+    def __init__(self):
+        self.values = {}
+
+    def set(self, name, value):
+        self.values[name] = value
+
+    def get(self, name):
+        return self.values.get(name)
+
+
+class TestTimer(TestCase):
+
+    def setUp(self):
+        patch.object(Timer, 'connect_to_dbus').start()
+        self.config = Config()
+        self.config.set('hour', '00')
+        self.config.set('min', '00')
+        self.timer = Timer(self.config)
+
+    def tearDown(self):
+        patch.stopall()
+
+    def test_time_parser_classic_format(self):
+        self.assertEqual(self.timer.time_parser('00:00'), ('00', '00'))
+        self.assertEqual(self.timer.time_parser('2:40'), ('2', '40'))
```

Comment:
please, put each of those tests into a separate method and follow BDD style of naming methods.

You might ask why? If somebody breaks the parsing, now they get "you broke it". If you have many small methods, they can see how many % of the functionality they broke. Plus having name for each case it makes obvious what is the purpose of the test. E.g. if your method is `test_minutes_are_in_range` and you try it with "0:-40", "0:20", "0:60", "0:80", it is really obvious. Another method can be similar for hours. Another one for parsing "17:00pm" and etc